<a href="https://colab.research.google.com/github/Anusuya0901/Data-analyst/blob/main/pro%202.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
df=pd.read_csv('/content/colorectal_cancer_prediction.csv')

# Task
Clean the `/content/colorectal_cancer_prediction.csv` dataset by inspecting its structure, handling missing values, and removing duplicates, then propose a data analysis project focused on understanding factors influencing colorectal cancer prediction.

## Inspect Data for Cleaning

### Subtask:
Examine the dataset to understand its structure, data types, and identify any missing values or potential inconsistencies that need cleaning. This will involve using functions like `df.info()`, `df.describe()`, `df.head()`, and `df.isnull().sum()`.


**Reasoning**:
To get an initial overview of the data and verify it loaded correctly, I will display the first few rows of the DataFrame using `df.head()` as per the first instruction.



In [ ]:
df.head()

Patient_ID  Age  Gender   Race         Region Urban_or_Rural  \
0           1   71    Male  Other         Europe          Urban   
1           2   34  Female  Black  North America          Urban   
2           3   80  Female  White  North America          Urban   
3           4   40    Male  Black  North America          Rural   
4           5   43  Female  White         Europe          Urban   

  Socioeconomic_Status Family_History Previous_Cancer_History  \
0               Middle            Yes                      No   
1               Middle             No                      No   
2               Middle             No                      No   
3                  Low             No                      No   
4                 High            Yes                      No   

  Stage_at_Diagnosis  ... Insurance_Coverage Time_to_Diagnosis  \
0                III  ...                Yes           Delayed   
1                  I  ...                 No            Timely   
2                III  ...                Yes            Timely   
3                  I  ...                Yes           Delayed   
4                III  ...                 No           Delayed   

  Treatment_Access Chemotherapy_Received  Radiotherapy_Received  \
0             Good                   Yes                     No   
1             Good                    No                    Yes   
2          Limited                    No                    Yes   
3          Limited                   Yes                     No   
4             Good                   Yes                     No   

  Surgery_Received Follow_Up_Adherence Survival_Status Recurrence  \
0               No                Good        Survived         No   
1              Yes                Poor        Deceased         No   
2              Yes                Good        Survived         No   
3              Yes                Poor        Deceased         No   
4              Yes                Poor        Deceased        Yes   

  Time_to_Recurrence  
0                 16  
1                 28  
2                 26  
3                 44  
4                 20  

[5 rows x 30 columns]

**Reasoning**:
As per the second instruction, I will print a concise summary of the DataFrame `df` using `df.info()` to check data types, non-null values, and memory usage.



In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89945 entries, 0 to 89944
Data columns (total 30 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Patient_ID               89945 non-null  int64  
 1   Age                      89945 non-null  int64  
 2   Gender                   89945 non-null  object 
 3   Race                     89945 non-null  object 
 4   Region                   89945 non-null  object 
 5   Urban_or_Rural           89945 non-null  object 
 6   Socioeconomic_Status     89945 non-null  object 
 7   Family_History           89945 non-null  object 
 8   Previous_Cancer_History  89945 non-null  object 
 9   Stage_at_Diagnosis       89945 non-null  object 
 10  Tumor_Aggressiveness     89945 non-null  object 
 11  Colonoscopy_Access       89945 non-null  object 
 12  Screening_Regularity     89945 non-null  object 
 13  Diet_Type                89945 non-null  object 
 14  BMI                   

**Reasoning**:
As per the third instruction, I will generate descriptive statistics for the numerical columns in `df` using `df.describe()` to understand their distribution and central tendencies.



In [ ]:
df.describe()

Patient_ID           Age           BMI  Time_to_Recurrence
count  89945.000000  89945.000000  89945.000000        89945.000000
mean   44973.000000     54.332892     29.268105           29.543299
std    25965.029318     20.182220      6.211668           17.268440
min        1.000000     20.000000     18.500000            0.000000
25%    22487.000000     37.000000     23.900000           15.000000
50%    44973.000000     54.000000     29.300000           30.000000
75%    67459.000000     72.000000     34.600000           44.000000
max    89945.000000     89.000000     40.000000           59.000000

**Reasoning**:
As per the fourth instruction, I will calculate and display the total number of missing values for each column in `df` using `df.isnull().sum()` to identify any missing data.



In [ ]:
df.isnull().sum()

Patient_ID                 0
Age                        0
Gender                     0
Race                       0
Region                     0
Urban_or_Rural             0
Socioeconomic_Status       0
Family_History             0
Previous_Cancer_History    0
Stage_at_Diagnosis         0
Tumor_Aggressiveness       0
Colonoscopy_Access         0
Screening_Regularity       0
Diet_Type                  0
BMI                        0
Physical_Activity_Level    0
Smoking_Status             0
Alcohol_Consumption        0
Red_Meat_Consumption       0
Fiber_Consumption          0
Insurance_Coverage         0
Time_to_Diagnosis          0
Treatment_Access           0
Chemotherapy_Received      0
Radiotherapy_Received      0
Surgery_Received           0
Follow_Up_Adherence        0
Survival_Status            0
Recurrence                 0
Time_to_Recurrence         0
dtype: int64

## Handle Missing Values

### Subtask:
Address any missing values identified in the previous step. This may involve imputation strategies (e.g., mean, median, mode for numerical columns, or mode for categorical columns) or deciding whether to drop rows or columns with a significant amount of missing data.


## Check and Remove Duplicates

### Subtask:
Identify and remove any duplicate rows in the dataset to ensure data integrity and prevent skewed analysis results.


**Reasoning**:
To begin identifying and removing duplicate rows, I will first check the total number of duplicate rows in the DataFrame `df` using `df.duplicated().sum()` as per the first instruction.



In [ ]:
print(f"Number of duplicate rows before removal: {df.duplicated().sum()}")

Number of duplicate rows before removal: 0


**Reasoning**:
Since no duplicate rows were found in the previous step, the next logical step according to the instructions is to verify that no duplicates remain, which will confirm the dataset's integrity regarding duplicates. After this verification, the subtask of handling duplicates will be complete.



In [ ]:
print(f"Number of duplicate rows after removal: {df.duplicated().sum()}")

Number of duplicate rows after removal: 0


## Propose Data Analyst Project

### Subtask:
Propose a data analysis project focused on understanding factors influencing colorectal cancer prediction, including potential analytical questions and relevant visualizations, based on the cleaned dataset.


The dataset `df` contains 89,945 entries and 30 columns. From `df.info()`, we observed that there are no missing values across all columns, which simplifies the cleaning process significantly. The columns consist of a mix of numerical and categorical data types:

*   **Numerical columns**: `Patient_ID`, `Age`, `BMI`, `Time_to_Recurrence`. `Patient_ID` is an identifier, `Age` and `BMI` are continuous numerical features, and `Time_to_Recurrence` is also numerical.
*   **Categorical columns**: `Gender`, `Race`, `Region`, `Urban_or_Rural`, `Socioeconomic_Status`, `Family_History`, `Previous_Cancer_History`, `Stage_at_Diagnosis`, `Tumor_Aggressiveness`, `Colonoscopy_Access`, `Screening_Regularity`, `Diet_Type`, `Physical_Activity_Level`, `Smoking_Status`, `Alcohol_Consumption`, `Red_Meat_Consumption`, `Fiber_Consumption`, `Insurance_Coverage`, `Time_to_Diagnosis`, `Treatment_Access`, `Chemotherapy_Received`, `Radiotherapy_Received`, `Surgery_Received`, `Follow_Up_Adherence`, `Survival_Status`, `Recurrence`.

From `df.describe()`, we can see the range and distribution of numerical columns:
*   **Age**: Ranges from 20 to 89 years, with a mean of approximately 54 years.
*   **BMI**: Ranges from 18.5 to 40, with a mean of approximately 29.27.
*   **Time_to_Recurrence**: Ranges from 0 to 59, with a mean of approximately 29.54. This column might be particularly important for predicting recurrence.

The absence of missing values and duplicates (as confirmed by `df.isnull().sum()` and `df.duplicated().sum()`) indicates that the dataset is clean and ready for analysis.

## Proposed Data Analysis Project: Factors Influencing Colorectal Cancer Outcomes

This project aims to investigate various factors influencing colorectal cancer prediction, focusing on patient characteristics, medical history, lifestyle, and treatment outcomes using the cleaned dataset.

### Analytical Questions, Techniques, Visualizations, and Potential Insights:

**1. What demographic and lifestyle factors are most associated with the stage of colorectal cancer at diagnosis?**

*   **Key Features**: `Age`, `Gender`, `Race`, `Region`, `Urban_or_Rural`, `Socioeconomic_Status`, `BMI`, `Smoking_Status`, `Alcohol_Consumption`, `Diet_Type`, `Physical_Activity_Level`, `Red_Meat_Consumption`, `Fiber_Consumption` vs. `Stage_at_Diagnosis`.
*   **Analysis Techniques**: Comparative analysis (e.g., ANOVA, Chi-square tests) to assess statistical differences in demographic and lifestyle factors across different stages. Correlation analysis for numerical features like `Age` and `BMI`.
*   **Visualizations**:
    *   **Bar charts**: To show the distribution of categorical factors (e.g., Gender, Race, Smoking Status) within each stage.
    *   **Box plots**: To compare numerical factors (e.g., Age, BMI) across different stages of diagnosis.
    *   **Stacked bar charts**: To illustrate the breakdown of `Stage_at_Diagnosis` by different categorical variables.
*   **Potential Insights**: Identify high-risk groups based on demographics and lifestyle, which can inform targeted screening and prevention programs.

**2. How do family history, previous cancer history, and screening regularity influence `Survival_Status` and `Recurrence`?**

*   **Key Features**: `Family_History`, `Previous_Cancer_History`, `Screening_Regularity`, `Colonoscopy_Access`, `Insurance_Coverage` vs. `Survival_Status` and `Recurrence`.
*   **Analysis Techniques**: Chi-square tests to determine the association between these categorical factors and `Survival_Status`/`Recurrence`. Descriptive statistics to summarize survival and recurrence rates based on these factors.
*   **Visualizations**:
    *   **Bar charts**: Comparing `Survival_Status` and `Recurrence` rates based on `Family_History`, `Previous_Cancer_History`, `Screening_Regularity`, `Colonoscopy_Access`, and `Insurance_Coverage`.
    *   **Grouped bar charts**: To show survival/recurrence within different categories.
*   **Potential Insights**: Understand the impact of genetic predisposition, past medical conditions, and preventive care on patient outcomes. This can highlight the importance of early detection and personalized risk assessments.

**3. What is the effectiveness of different treatment modalities (`Chemotherapy_Received`, `Radiotherapy_Received`, `Surgery_Received`) on `Survival_Status` and `Recurrence`, considering `Stage_at_Diagnosis` and `Tumor_Aggressiveness`?**

*   **Key Features**: `Chemotherapy_Received`, `Radiotherapy_Received`, `Surgery_Received`, `Stage_at_Diagnosis`, `Tumor_Aggressiveness`, `Treatment_Access`, `Follow_Up_Adherence` vs. `Survival_Status` and `Recurrence`.
*   **Analysis Techniques**: Multivariate analysis (e.g., logistic regression or survival analysis if time-to-event data were more granular) to evaluate the combined and individual effects of treatment modalities adjusted for stage and aggressiveness. Comparative analysis (Chi-square) for categorical outcomes.
*   **Visualizations**:
    *   **Heatmaps**: To show the interaction between treatment types, stage, and outcomes.
    *   **Grouped bar charts**: To visualize `Survival_Status` and `Recurrence` across different treatment combinations, stratified by `Stage_at_Diagnosis` or `Tumor_Aggressiveness`.
*   **Potential Insights**: Provide evidence for the efficacy of various treatment strategies, guiding clinical decisions and improving patient care protocols. Identify optimal treatment paths for different stages and tumor characteristics.

**4. Is there a relationship between `Time_to_Diagnosis` and `Time_to_Recurrence` with patient outcomes (`Survival_Status`, `Recurrence`)?**

*   **Key Features**: `Time_to_Diagnosis`, `Time_to_Recurrence` vs. `Survival_Status`, `Recurrence`.
*   **Analysis Techniques**: Correlation analysis between `Time_to_Recurrence` and other factors. Comparative analysis (e.g., t-tests or ANOVA) to see if `Time_to_Recurrence` differs significantly between `Survival_Status` groups. Similarly for `Time_to_Diagnosis`.
*   **Visualizations**:
    *   **Histograms/KDE plots**: To visualize the distribution of `Time_to_Recurrence` and `Time_to_Diagnosis` for different outcome groups.
    *   **Scatter plots**: If a continuous variable could be related to `Time_to_Recurrence`.
    *   **Box plots**: To compare `Time_to_Recurrence` values based on `Survival_Status` or `Recurrence`.
*   **Potential Insights**: Understand the impact of diagnostic delays and recurrence timing on overall prognosis. This could highlight areas for improving diagnostic pathways and post-treatment monitoring.

## Final Task

### Subtask:
Summarize the data cleaning steps performed and present the proposed data analysis project, outlining the next steps for analysis and visualization.


## Summary:

### Data Analysis Key Findings

*   The dataset contains 89,945 entries and 30 columns, encompassing patient demographics, medical history, lifestyle factors, treatment details, and outcome variables.
*   The dataset is remarkably clean, with **no missing values** identified across any columns and **no duplicate rows** found, indicating high data integrity.
*   Columns include 3 integers (`Patient_ID`, `Age`, `Time_to_Recurrence`), 1 float (`BMI`), and 26 object-type columns, primarily representing categorical data.
*   Numerical features show `Age` ranging from 20 to 89 years (mean \~54), `BMI` from 18.5 to 40 (mean \~29.27), and `Time_to_Recurrence` from 0 to 59 (mean \~29.54).
*   A comprehensive data analysis project has been proposed, focusing on understanding factors influencing colorectal cancer outcomes. It includes four analytical questions covering:
    1.  Demographic and lifestyle factors associated with `Stage_at_Diagnosis`.
    2.  Influence of family history, previous cancer, and screening regularity on `Survival_Status` and `Recurrence`.
    3.  Effectiveness of treatment modalities considering `Stage_at_Diagnosis` and `Tumor_Aggressiveness`.
    4.  Relationship between `Time_to_Diagnosis`, `Time_to_Recurrence`, and patient outcomes.

### Insights or Next Steps

*   The exceptional cleanliness of the dataset regarding missing values and duplicates significantly streamlines the preprocessing phase, allowing for immediate progression to feature engineering and modeling.
*   The next critical step is to preprocess the 26 object-type (categorical) columns using appropriate encoding techniques (e.g., one-hot encoding or label encoding) to prepare the data for statistical analysis and machine learning models, as outlined in the proposed project.
